projects/admin/src/app/acquisition/api/acq-order-api.service.ts

```diff
@@ -113,6 +117,30 @@ export class AcqOrderApiService extends BaseApi {
     );
   }
 
+  getOrderHistory(orderPid: string) {
+    // check if orderPid is already present into the previously loaded history items.
+    // If YES :: not needed to reload the history, just update the `current` attribute of history items
+    const orderRef = new URL(this._apiService.getRefEndpoint('acq_orders', orderPid));
+    const idx = this._acqOrderHistory.findIndex(item => item.$ref === orderRef.toString());
+    if (idx !== -1) {
+      if (!this._acqOrderHistory[idx].current) {
+        this._acqOrderHistory.map(version => version.current = false);
+        this._acqOrderHistory[idx].current = true;
+        this.acqOrderHistorySubject.next(this._acqOrderHistory);
+      }
+    } else {
+      // If NO :: Load the acquisition order history
+      const apiUrl = `/api/acq_order/${orderPid}/history`;
+      return this._http
+        .get<AcqOrderHistoryVersionResponseInterface[]>(apiUrl)
+        .subscribe(versions => {
```

Comment:
`subscribe` or `pipe`?

Reply:
seems same behavior here. The important point is that is update the "acqOrderHistorySubject". This is this subject that must be subscribed.
I will remove the return statement.... nonsense here.